docs/root/start/sandboxes/csrf.rst

```diff
@@ -0,0 +1,105 @@
+on two different virtual machines with different origins.
+
+The frontend has a field to input the remote domain of where you would like to
+send POST requests along with radio buttons to select the remote domain's CSRF
+enforcement. The CSRF enforcement choices are:
+
+  * Disabled: CSRF is disabled on the requested route. This will result in a
+    successful request since there is no CSRF enforcement.
+  * Shadow Mode: CSRF is not enforced on the requested route but will record
+    if the request contains a valid source origin.
+  * Enabled: CSRF is enabled and will return a 403 (Forbidden) status code when
+    a request is made from a different origin.
+  * Ignored: CSRF is enabled but the request type is a GET. This should bypass
+    the CSRF filter and return successfully.
+
+Running the Sandboxes
+~~~~~~~~~~~~~~~~~~~~~
+
+The following documentation runs through the setup of both services.
+
+**Step 1: Install Docker**
+
+Ensure that you have a recent versions of ``docker`` and ``docker-compose``.
+
+A simple way to achieve this is via the `Docker Toolbox <https://www.docker.com/products/docker-toolbox>`_.
+
+**Step 2: Clone the Envoy repo and start all of our containers**
+
+If you have not cloned the Envoy repo, clone it with ``git clone git@github.com:envoyproxy/envoy``
+or ``git clone https://github.com/envoyproxy/envoy.git``
+
+Terminal 1
+
+.. code-block:: console
+
+  $ pwd
+  envoy/examples/csrf
+  $ docker-compose pull
+  $ docker-compose up --build -d
+  $ docker-compose ps
+
+            Name                        Command          State                            Ports
+  ----------------------------------------------------------------------------------------------------------------------
+  csrf_front-envoy_1      /docker-entrypoint.sh /bin ... Up      10000/tcp, 0.0.0.0:8000->80/tcp, 0.0.0.0:8001->8001/tcp
+  csrf_service_1          /bin/sh -c /usr/local/bin/ ... Up      10000/tcp, 80/tcp
+
+Terminal 2
+
+.. code-block:: console
+
+  $ pwd
+  envoy/examples/csrf
+  $ docker-compose up --build -d
+  $ docker-compose ps
+
+            Name                       Command           State                            Ports
+  ----------------------------------------------------------------------------------------------------------------------
+  csrf_front-envoy_1      /bin/sh -c /usr/local/bin/ ... Up      10000/tcp, 0.0.0.0:8000->80/tcp, 0.0.0.0:8001->8001/tcp
+  csrf_service_1          /docker-entrypoint.sh /bin ... Up      10000/tcp, 80/tcp
+
+**Step 3: Test Envoy's CSRF capabilities**
+
+You can now open a browser to view your frontend service at ``localhost:8000``.
+
+Results of the cross-site request will be shown on the page under *Request Results*.
+Your browser's CSRF enforcement logs can be found in the console and in the
+network tab.
+
+For example:
+
+.. code-block:: console
+
+  Failed to load resource: the server responded with a status of 403 (Forbidden)
```

Comment:
In my testing on macOS with `localhost` in the field all 4 radio buttons (Disabled, Shadow Mode, Enabled and Ignored) all report "Success!". If I change the input field to `127.0.0.1` the Enabled radio button only returns "Rejected by CSRF".

Reply:
Cleaned things up a bit. I hope that makes things clearer.